oslc4j-jena-provider/src/main/java/org/eclipse/lyo/oslc4j/provider/jena/JenaModelHelper.java

```diff
@@ -525,6 +522,7 @@ public static void skolemize(final Model m, Function<BlankNodeId, String> skolem
 			final Map<Class<?>, Map<String, Method>> classPropertyDefinitionsToSetMethods = new HashMap<>();
             Class<?> originalBeanClass = beanClass;
 			for (final Resource resource : listSubjects) {
+                beanClass = originalBeanClass;
```

Comment:
What’s this?

Reply:
It seems to me that it is reassigning the reference of the original   `beanClass` value in each iteration, because this could be changed by the `ResourcePackages.getClassOf` method, depending to the `resource` type. Is that right @neormx ?

Maybe this should be added at the bottom of the `for`.

Reply:
Correct, while iterating resources, the reference to the beanClass can be changed by the most concrete reference, we need to reset to the originalBeanClass in order to process next resource in case there are more than one to unmarshall in model. This is a bug I wanted to fix along with this PR.